tests/unit/tasks/generate-from-blueprint-test.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const GenerateFromBlueprintTask = require('../../../lib/tasks/generate-from-blueprint');
+const MockProject = require('../../helpers/mock-project');
+const expect = require('../../chai').expect;
+
+describe('GenerateFromBlueprintTask', function() {
+  let task;
+  beforeEach(function() {
+    task = new GenerateFromBlueprintTask({
+      project: new MockProject(),
+    });
+  });
+
+  describe('prefixBlueprintName', function() {
+
+    describe('in a non module unification app', function() {
+      it('does not prefix `component`', function() {
+        let name = task.prefixBlueprintName('component');
+        expect(name).to.equal('component');
+      });
+    });
+
+    describe('in a module unification app', function() {
+      beforeEach(function() {
+        task.project.isModuleUnification = function() {
+          return true;
+        };
+      });
+
+      it('prefixes `component`', function() {
+        let name = task.prefixBlueprintName('component');
+        expect(name).to.equal('module-unification-component');
```

Comment:
Should this not still be `component`?
We could probably hook into `isModuleUnification` that you added  and override `__path__` and `__templatepath__` in [ember.js component blueprint](https://github.com/emberjs/ember.js/blob/master/blueprints/component/index.js#L26..L43)

Reply:
It could be, I considered that approach. I think it's cleaner to keep these blueprints separate though, especially given that there isn't much reuse between the contents of the files

Reply:
A downside of my approach is that there is a coupling between ember-cli and ember (you would need to update both to use the new MU blueprints). Your suggested approach doesn't have this issue.

I'll try your approach as see how it sits, thanks!

Reply:
> would need to update both to use the new MU blueprints and again when we rename ..

I wouldn't say that's a downside to your approach, as we could pollyfill the MU blueprints, and issue a deprecation warning when the hosts' app version doesn't need the polyfill anymore

Reply:
true

Reply:
I think the big difference is that we can experiment and build blueprints with `isModuleUnification` in ember (or even addons) without having to enable each different type of MU-blueprint here https://github.com/ember-cli/ember-cli/pull/7541/files#diff-c3643df1b2f6f7f28a101393e4fe0c12R99 

Already we can support many permutations of the same file type like we do here https://github.com/emberjs/ember.js/tree/master/blueprints/controller-test (e.g we could have a `module-unification-files` prefix there)

Reply:
👍

I don't believe that we've got access to the project in the blueprints, I'll add a `options.moduleUnification` property and refactor the `component` blueprint

Reply:
here it is: https://github.com/emberjs/ember.js/blob/master/blueprints/component/index.js#L58

Reply:
great, thanks

Reply:
thanks for your feedback, updated